### PR TITLE
Issue13

### LINKED ISSUE #13: Add JSON route for Projects collection
Create a JSON route that can be used to export documents from the Projects collection.
# User story

```
As a developer
I would like to access a JSON data endpoint for the Projects collection
so that I can integrate the data into my application
```
# Task

To complete this task, define at least two JSON routes:
- All projects
- Single project

It may be necessary to install a package to help with the JSON routing. To this end, the following packages have been mentioned:
- [simple:rest](https://atmospherejs.com/simple/rest)
- [simple:json-routes](https://atmospherejs.com/simple/json-routes)

Based on discussion, we will try the **simple:json-routes** first.
## All projects

Route: `/projects` 

The all projects route should return a JSON array of all projects, with each project document containing all project details.
## Single project

Route: `/projects/{{ _id }}`

The route should allow users to request a specific Project ID, and receive a JSON document with .Project details-
# Acceptance criteria
- [x] "All projects" route returns a JSON array of all projects
  - [x] Each project in array contains all current project details
- [x] "Single project" route returns a JSON object for one project
  - [x] Route uses ID parameter to identify project
  - [x] JSON document returns all current details about requested project

### PR DESCRIPTION
**Pull request by 2ECC71**
Adds two json routes, /projects/:id and /projects
This closes #13

### CODE REVIEW

File: projects/router/routes.js [hunk 1]
```diff
@@ -9,3 +9,14 @@ FlowRouter.route("/projects/add", {
     BlazeLayout.render("mainLayout", {main: "addProject"});
   }
 });
+
+JsonRoutes.add("get", "/projects/:id", function (req, res, next) {
+  var id = req.params.id;
```
Comment (brylie) [inline projects/router/routes.js]: Please use a more descriptive variable name here. E.g. `projectId`

This is a good habit to start, as it improves code readability, particularly in complex files.

File: projects/router/routes.js [hunk 2]
```diff
@@ -9,3 +9,14 @@ FlowRouter.route("/projects/add", {
...
+
+  JsonRoutes.sendResult(res, 200, Projects.findOne(id));
+});
+
+JsonRoutes.add("get", "/projects", function (req, res, next) {
+
+  JsonRoutes.sendResult(res, 200, Projects.find().fetch());
```
Comment (brylie) [inline projects/router/routes.js]: Please do the `Projects.find().fetch()` in a separate step, and include a semantic variable name such as `allProjects`.

File: projects/router/routes.js [hunk 3]
```diff
@@ -9,3 +9,14 @@ FlowRouter.route("/projects/add", {
...
 });
+
+JsonRoutes.add("get", "/projects/:id", function (req, res, next) {
+  var id = req.params.id;
+
+  JsonRoutes.sendResult(res, 200, Projects.findOne(id));
+});
+
+JsonRoutes.add("get", "/projects", function (req, res, next) {
+
```
Comment (brylie) [inline projects/router/routes.js:24]: Add a comment here describing the general intention of this code in human language. While the JavaScript may be relatively easy to read, providing a human language description of the intention has the following benefits:
- Clearly explains the purpose of code to other developers or your future self
  - comments act like an 'async pair programmer'
- doesn't require a human to parse JavaScript, unless they are debugging/enhancing the code
- provides an opportunity to reflect on the purpose of the code, possibly illuminating logical flaws
  - e.g. variable names that don't reflect their reference
  - logical errors in the line of code

File: projects/router/routes.js (hunk 1 above)
Comment (brylie) [inline projects/router/routes.js]: Add a comment here describing the general intention of this code in human language.

**Comment by 2ECC71** [comment]
How would I do it in two steps?
var allProjects = Projects.find().fetch();
JsonRoutes.sendResult(res, 200, allProjects);
?

**Comment by brylie** [comment]
Yep. Basically, the code does two things anyway:
- Gets an array of all projects
- sends an HTTP response

So, separating the two steps would fall along those lines, with semantic variable naming and human-language comment for each line.

**Comment by brylie** [comment]
This looks good. Go ahead and merge :-)

**Comment by brylie** [comment]
Really great inline comments.

**Comment by brylie** [comment]
Also, you can title your PRs with the issue number and it will automatically close the issue.

E.g. "Closes #13"

**Comment by brylie** [comment]
https://help.github.com/articles/closing-issues-via-commit-messages/

**Comment by 2ECC71** [comment]
Ok, thanks!

**Pull request merged**